src/Type/Definition/EnumType.php

```diff
@@ -88,9 +94,9 @@ private function getNameLookup() : ArrayObject
     /**
      * @return EnumValueDefinition[]
      */
-    public function getValues()
+    public function getValues() : array
     {
-        if ($this->values === null) {
+        if (! isset($this->values)) {
```

Comment:
Could `values` be initialized to an empty array at declaration time? Or can it legitimately be an empty array even after lazy load?

Reply:
If it were to become an empty array after initialization, and we check for `count($this->values) === 0`, we would re-initialize unnecessarily. I think such an Enum type would be invalid, but i see nothing that stops a `$config` with empty `values` being passed.

The way it is written now clearly expresses that we run initialize `$this->values` only if it is not set, no matter what it is set to afterwards.

Reply:
> If it were to become an empty array after initialization

That's exactly what I'm asking. If that is a possibility, then what you have now is great. But if empty array is impossible post-lazy load, then you could try the other way. But it sounds like we're not exactly sure, so I suppose it's safest to go with what you have.